[Docs] Add permission setup page for the clouds

Michaelvll opened the pull request:
<!-- Describe the changes in this PR -->

This PR adds the instructions for set up the cloud accounts with the required permissions.

Future TODO:
- [ ] Add instruction for Azure.

<!-- Describe the tests ran -->
<!-- Unit tests (tests/test_*.py) are part of GitHub CI; below are tests that launch on the cloud. -->

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - rendered locally
- [ ] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
Move this to the bottom of USER GUIDES? We can reserve "References" for CLI/API/YAML refs.

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -0,0 +1,195 @@
+
+AWS accounts can be attached with a policy that limits the permissions of the account. Follow these steps to create an AWS user with the minimum permissions required by SkyPilot:
+
+1. Open the IAM dashboard in the AWS console and click on the "Users" tab. Then, click "Add users" and enter the user's name.
```
For the whole doc: for referring to buttons that appear in cloud consoles, e.g., `"Users"`, can we bold it `**Users**`? Following a common doc styleguide.

concretevitamin commented on docs/source/getting-started/installation.rst:
```diff
@@ -62,8 +64,11 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
   $ # Configure your AWS credentials
   $ aws configure
 
+To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
```
Can we add something like **Optional** and/or "not needed to get started with SkyPilot" disclaimer?

I am worried that it sounds like required steps.

Perhaps put it below L69.

Same for GCP?

concretevitamin commented:
We may want to merge this soon :)

Michaelvll commented:
Sorry for missing the reviews @concretevitamin! Just updated the doc and fixed all the warnings for the doc building. Please take another look. Thanks!

concretevitamin commented on docs/source/getting-started/installation.rst:
```diff
@@ -66,13 +66,12 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
   $ aws configure
 
-To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
+Note: If you are using AWS IAM Identity Center (AWS SSO), you will need :code:`pip install awscli>=1.27.10`. See `here <https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-sso.html>`__ for instructions on how to configure AWS SSO.
```
Merge with master? https://skypilot.readthedocs.io/en/latest/getting-started/installation.html#amazon-web-services-aws

Michaelvll replied:
Oops, done. Thanks!

concretevitamin commented on docs/source/index.rst:
```diff
@@ -103,6 +103,7 @@ Documentation
    reference/quota
    reference/logging
    reference/faq
+   reference/cloud-permissions
```
Thinking we should probably add a new section with one of these names:
- Cloud Administration
- Cloud Administration and Usage
- Cloud Setup and Usage
- better name

?

And then we can put
- cloud-permissions: Minimal Cloud Permissions Setup
- cloud-administration: Using Cloud-specific Features
  - may want to rename the filename too 
- quota: Requesting Quota

under it. Note I put some renamed titles above. How do they read?

Michaelvll replied:
Good point. I also re-ordered the "User Guides" a bit to de-emphasize the local mode. I also chose to use shorter name to make sure they can display in one line in the content table.

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -15,23 +20,25 @@ AWS
 AWS accounts can be attached with a policy that limits the permissions of the account. Follow these steps to create an AWS user with the minimum permissions required by SkyPilot:
 
-1. Open the IAM dashboard in the AWS console and click on the "Users" tab. Then, click "Add users" and enter the user's name.
+1. Open the `IAM dashboard <https://us-east-1.console.aws.amazon.com/iamv2/home#/home>_` in the AWS console and click on the **Users** tab. Then, click **Add users** and enter the user's name. Click **Next**.
```
url rendered

Michaelvll replied:
Oops, fixed. Thanks!

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -100,30 +107,36 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
     :alt: AWS Add Policy
-7. [Optional] If you would like to have your users access the s3 bucket. You can additionally attach the S3 access, such as the "AmazonS3FullAccess" policy.
+
+6. [Optional] If you would like to have your users access the s3 bucket. You can additionally attach the S3 access, such as the "AmazonS3FullAccess" policy.
```
```suggestion
6. [Optional] If you would like to have your users access S3 buckets: You can additionally attach S3 access, such as the "AmazonS3FullAccess" policy.
```

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -100,30 +107,36 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
 
 With the steps above you are almost ready to have the users in your organization to use SkyPilot with the minimal permissions.
 
 **One more thing** to do is to create a single service account "skypilot-v1" for all the users in your organization. There are two ways to accomplish this:
```
```suggestion
**One more thing** to do is to create a single service account ``skypilot-v1`` for all users in your organization. There are two ways to accomplish this:
```

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -146,11 +159,16 @@ With the steps above you are almost ready to have the users in your organization
             }
-2. Instead, you can create the "skypilot-v1" service account manually. The following section describe how to create the service account manually.
+
+2. Alternatively, you can create the "skypilot-v1" service account manually. The following section describe how to create the service account manually.
```
```suggestion
2. Alternatively, you can create the ``skypilot-v1`` service account manually. The following section describes how to create the service account manually.
```

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -146,11 +159,16 @@ With the steps above you are almost ready to have the users in your organization
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. note::
+
+    If you already have a service account called "skypilot-v1" in your AWS account, it is likely created by SkyPilot automatically, and you can skip this section.
```
```suggestion
    If you already have a service account called ``skypilot-v1`` in your AWS account, it is likely created by SkyPilot automatically, and you can skip this section.
```

concretevitamin commented on docs/source/reference/cloud-permissions.rst:
```diff
@@ -100,30 +107,36 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
 
 With the steps above you are almost ready to have the users in your organization to use SkyPilot with the minimal permissions.
 
 **One more thing** to do is to create a single service account "skypilot-v1" for all the users in your organization. There are two ways to accomplish this:
```
Should we replace "service account" with "IAM role" in this doc? This is the AWS terminology (i.e., I found the role in https://us-east-1.console.aws.amazon.com/iamv2/home?region=us-east-1#/roles).

Michaelvll replied:
Good point! Changed.

concretevitamin reviewed (commented):
This is great @Michaelvll! Some notes.

concretevitamin commented on docs/source/getting-started/installation.rst:
```diff
@@ -66,6 +68,8 @@ To get the **AWS access key** required by :code:`aws configure`, please go to th
 
 To use AWS IAM Identity Center (AWS SSO), see :ref:`here<aws-sso>` for instructions.
 
+[**Optional**] To create a new AWS user with minimal permissions for SkyPilot, see :ref:`AWS User Creation <cloud-permissions-aws>`.
```
Nit: Can we `[**Optional**]` -> `**Optional**:`? Looks better to me. Same for GCP below?

Michaelvll replied:
Sounds good. Updated.

concretevitamin commented on docs/source/index.rst:
```diff
@@ -96,14 +96,12 @@ Documentation
-   reference/cloud-administration
-   reference/quota
-   reference/logging
+   reference/clouds-setup/index
```
In previous comment I was more thinking of adding a section parallel to User Guides:
```
.. toctree::
   :maxdepth: 1
   :caption: Cloud Administration and Usage
```
to hold these 3 pages. This way, an "administrator" will easily see that "oh they have docs for me".

Wdyt?

Michaelvll replied:
That also makes sense to me. Moved. Thanks!

concretevitamin commented on docs/source/reference/clouds-setup/cloud-permissions.rst:
```diff
@@ -127,9 +127,9 @@ AWS accounts can be attached with a policy that limits the permissions of the ac
 With the steps above you are almost ready to have the users in your organization to use SkyPilot with the minimal permissions.
 
-**One more thing** to do is to create a single service account "skypilot-v1" for all the users in your organization. There are two ways to accomplish this:
+**One more thing** to do is to create a single iam role ``skypilot-v1`` for all users in your organization. There are two ways to accomplish this:
```
s/iam/IAM everywhere?

Michaelvll replied:
Good catch! Done. Thanks!

concretevitamin commented on docs/source/reference/local/index.rst:
```diff
@@ -3,6 +3,10 @@
 
+.. note::
+
+   This feature has been **DEPRECATED**, and under the process of re-implementing. Please stay tuned for updates.
```
```suggestion
   This feature has been **DEPRECATED**, and is under the process of re-implementation. Please stay tuned for updates.
```

Michaelvll replied:
Oops. Good catch! Fixed. Thanks!

Michaelvll commented on docs/source/reference/clouds-setup/cloud-login.rst:
```diff
@@ -1,7 +1,7 @@
 
-Cloud Administration
-=============================
+Using Cloud-Specific Login
```
I like the "Authentication" better. Changed it and removed the L7-8 as the parent index has been removed.

Michaelvll commented on docs/source/reference/clouds-setup/index.rst:
```diff
@@ -0,0 +1,14 @@
+.. _cloud-setup:
+
+Advanced Cloud Setup
```
Removed index. Thanks!

concretevitamin reviewed (approved):
Thanks a lot for adding these docs @Michaelvll!
